dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SharingController.java

```diff
@@ -496,4 +506,22 @@ private void syncSharingForEventProgram( Program program )
         programStage.setCreatedBy( program.getCreatedBy() );
         manager.update( programStage );
     }
+
+    private void syncSharingForExpressionDimensionItems( Visualization visualization,
+        List<SharingUserAccess> sharingUserAccesses )
+    {
+        List<ExpressionDimensionItem> expressionDimensionItems = visualization.getDataDimensionItems()
+            .stream()
+            .map( DataDimensionItem::getExpressionDimensionItem )
+            .filter( Objects::nonNull )
+            .collect( Collectors.toList() );
+
+        expressionDimensionItems.forEach( edi -> {
+            org.hisp.dhis.user.sharing.Sharing sharing = edi.getSharing();
+            Set<UserAccess> userAccess = sharingUserAccesses.stream()
+                .map( sua -> new UserAccess( sua.getAccess(), sua.getId() ) ).collect( Collectors.toUnmodifiableSet() );
+            sharing.setUserAccesses( userAccess );
+            manager.update( edi );
```

Comment:
- Could `ExpressionDimensionItem` be shared to `UserGroup` ? If yes then we need to do the same for `SharingUserGroupAccess` here.

- How big is the size of `expressionDimensionItems` list ? Maybe consider to write a separate method `update ExpressionDimensionItem( List< ExpressionDimensionItem> items)`.  just to avoid sending too many update queries separately to database. 

- Do we need to implement this logic in the  `api/metadata` endpoint ?  User can import `Dashboard` or `Visualization` with `sharing` settings in same payload through metadata import service.  If yes then we need to add this logic to `VisualizationObjectBundleHook`.